core/domain/exp_services.py

```diff
@@ -557,6 +557,9 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
         exploration, exp_versions_diff=exp_versions_diff,
         revert_to_version=None)
 
+def save_exploration_test_helper(
+    committer_id, exploration, commit_message, change_list):
+    _save_exploration(committer_id, exploration, commit_message, change_list)
```

Comment:
This is a pattern that we'd want to avoid. Try this instead:
- Is this function being used by a class other than this one? If so, just make this function public.
- Is this function a good candidate to make public? If so, make this function public. (But make sure there's a good reason to do so).
- Is this function being called by another method within the class? If so, add a testing flag to that function that defaults to a False boolean when not in 'testing mode" that calls this private method directly.

Reply:
Hi @kevinlee12. Thanks for the advice! Looking at the references of `_save_exploration()`, I found that it is being used in multiple functions. Also, it is a behavioural test right? 
So it can be made public right?

Reply:
It depends where `_save_exploration()` is being referenced, if it's being referenced in other classes that are not tests, then it's definitely a candidate for flipping it to public.

Reply:
It is being called in `exp_services.update_exploration()` and some tests.

Reply:
So it looks like it’s just this class, so it’s probably best to just put a test flag in `update_exploration`.

Reply:
Umm, okay. But, what to do with the calls to `_save_exploration()` in the tests?

Reply:
We typically don't consider calls to private functions in tests for this scenario. The reason is that it's usually an artificial set up and therefore we would be turning the function public for the sake of completing the ticket. As for the flag approach, see https://github.com/oppia/oppia/pull/7837.

Reply:
Okay, so you mean I should make `_save_exploration()` public and use a 'testing' flag in `update_exploration()` right? Also, will have to pass the testing flag as `True` whenever it is called in a test.

Reply:
correct. just remember to default it to `False` so that we don't have to worry about it during normal circumstances.

Reply:
Okay!